kythe/java/com/google/devtools/kythe/extractors/java/bazel/JavaExtractor.java

```diff
@@ -200,16 +208,25 @@ public InputStream openStream() throws IOException {
     return files;
   }
 
-  /** Reads Bazel's compilation params file and returns the value of the --sourcegendir flag. */
+  private static final String SOURCEGENDIR_FLAG = "--sourcegendir";
+
+  /** Reads Bazel's compilation parameters and returns the value of the --sourcegendir flag. */
   private static Optional<Path> readGeneratedSourceDirParam(JavaCompileInfo jInfo)
       throws IOException {
+    for (int i = 0; i < jInfo.getArgumentCount() - 1; i++) {
```

Comment:
Can this flag appear more than once? What is java's behavior if it appears twice, does it take the first or the last?

Reply:
Not here, the parameters are controlled by Bazel itself.